src/test/java/org/prebid/server/bidder/grid/GridBidderTest.java

```diff
@@ -83,9 +94,93 @@ public void makeHttpRequestsShouldReturnErrorIfExtImpGridNull() {
-        assertThat(result.getErrors().get(0).getType()).isEqualTo(BidderError.Type.bad_input);
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getExt)
+                .containsExactly(
+                        mapper.valueToTree(ExtImpGrid.builder()
+                                .data(ExtImpGridData.of("pbadslot",
+                                        ExtImpGridDataAdServer.of("name", "adslot")))
+                                .bidder(ExtImpGridBidder.of(1, null))
+                                .gpid("adslot")
+                                .build()));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCorrectlyModifyRequestExt() throws JsonProcessingException {
+        // given
+        final ObjectNode impUserKeywordsNode = (ObjectNode) mapper.readTree(
+                "{\"firstPublisher\":[{\"name\":\"firstKeywordsUserSection\","
+                        + "\"segments\":[{\"name\":\"segment1\",\"value\":\"value1\"}]}]}");
+        final ObjectNode impSiteKeywordsNode = (ObjectNode) mapper.readTree(
+                "{\"firstPublisher\":[{\"name\":\"firstKeywordsSiteSection\","
+                        + "\"segments\":[{\"name\":\"segment1\",\"value\":\"value1\"}]}]}");
+        final ExtImpGrid impExt = ExtImpGrid.builder()
+                .data(ExtImpGridData.of("pbadslot",
+                        ExtImpGridDataAdServer.of("name", "adslot")))
+                .bidder(ExtImpGridBidder.of(1, Keywords.of(impUserKeywordsNode, impSiteKeywordsNode)))
+                .build();
+
+        final ExtRequest extRequest = ExtRequest.of(null);
+        extRequest.addProperty("keywords", mapper.readTree(
+                "{\"user\": {\"secondPublisher\":[{\"name\":\"secondKeywordsUserSection\","
+                        + "\"segments\":[{\"name\":\"segment2\",\"value\":\"value2\"}]}]}, "
+                        + "\"site\": {\"secondPublisher\":[{\"name\":\"secondKeywordsSiteSection\","
+                        + "\"segments\":[{\"name\":\"segment2\",\"value\":\"value2\"}]}]}}"));
+
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder().ext(mapper.valueToTree(impExt)).build()))
+                .id("request_id")
+                .ext(extRequest)
+                .site(Site.builder().keywords("siteKeyword1,siteKeyword2").build())
+                .user(User.builder().keywords("userKeyword1,userKeyword2").build())
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = gridBidder.makeHttpRequests(bidRequest);
+
+        // then
+        final String expectedRequestExtJson =
+                "{\"user\":{\"ortb2\":[{\"name\":\"keywords\",\"segments\":[{\"name\":\"keywords\","
```

Comment:
Seems like node approach makes too complex testing and also we don't have much logic from go like consistent ordering for publisher item map, we don't have all keywords etc. and others. So what if we will try work with map(maybe TreeMap for ordering) and use as much models in possible places as we can, and only for specific cases use nodes, what do you think?